rpc/src/server/router.rs

```diff
@@ -7,87 +7,92 @@ use std::sync::Arc;
+    routes.handle(vec![Method::GET], "/chains/:chain_id/blocks/:block_id/metadata", shell_handler::chains_block_id_metadata);
+    routes.handle(vec![Method::POST], "/injection/operation", shell_handler::inject_operation);
+    routes.handle(vec![Method::GET], "/workers/prevalidators", shell_handler::worker_prevalidators);
+    routes.handle(vec![Method::GET], "/config/network/user_activated_upgrades", shell_handler::config_user_activated_upgrades);
+    routes.handle(vec![Method::GET], "/config/network/user_activated_protocol_overrides", shell_handler::config_user_activated_protocol_overrides);
     // TODO: TE-174: just for sandbox
     if is_sandbox {
-        routes.handle(Method::POST, "/injection/block", shell_handler::inject_block);
+        routes.handle(vec![Method::POST], "/injection/block", shell_handler::inject_block);
     }
 
     // Shell rpcs - routed through ffi calls
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/preapply/operations", shell_handler::preapply_operations);
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/preapply/block", shell_handler::preapply_block);
+    routes.handle(vec![Method::POST], "/chains/:chain_id/blocks/:block_id/helpers/preapply/operations", shell_handler::preapply_operations);
+    routes.handle(vec![Method::POST], "/chains/:chain_id/blocks/:block_id/helpers/preapply/block", shell_handler::preapply_block);
 
     // Protocol rpcs - implemented
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/context/constants", protocol_handler::context_constants);
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/baking_rights", protocol_handler::baking_rights);
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/helpers/endorsing_rights", protocol_handler::endorsing_rights);
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/votes/listings", protocol_handler::votes_listings);
+    routes.handle(vec![Method::GET], "/chains/:chain_id/blocks/:block_id/context/constants", protocol_handler::context_constants);
+    routes.handle(vec![Method::GET], "/chains/:chain_id/blocks/:block_id/helpers/baking_rights", protocol_handler::baking_rights);
+    routes.handle(vec![Method::GET], "/chains/:chain_id/blocks/:block_id/helpers/endorsing_rights", protocol_handler::endorsing_rights);
+    routes.handle(vec![Method::GET], "/chains/:chain_id/blocks/:block_id/votes/listings", protocol_handler::votes_listings);
 
     // Other Protocol rpcs - routed through ffi calls
-    // TODO: rework this, this should accept more methods
-    routes.handle(Method::GET, "/chains/:chain_id/blocks/:block_id/*any", protocol_handler::call_protocol_rpc);
+    routes.handle(vec![Method::GET, Method::POST, Method::OPTIONS, Method::PUT], "/chains/:chain_id/blocks/:block_id/*any", protocol_handler::call_protocol_rpc);
```

Comment:
@adonagy 
`Method::OPTIONS` better change to: `Method::OPTIONS` -> `Method::DELETE`